classes/Tools.php

```diff
@@ -4228,6 +4230,45 @@ public static function isFileFresh(string $file, int $timeout = self::CACHE_LIFE
 
         return false;
     }
+
+    /**
+     * @return bool
+     */
+    public static function isCountryFromBrowserAvailable(): bool
+    {
+        $languageAvailable = static::getCountryIsoCodeFromHeader();
+        if ($languageAvailable === null) {
+            return false;
+        }
+
+        return Configuration::get('PS_DETECT_COUNTRY')
+            && static::isCountryIsoCodeAvailableInHeader()
+            && Validate::isLanguageIsoCode($languageAvailable);
+    }
+
+    /**
+     * @return bool
+     */
+    public static function isCountryIsoCodeAvailableInHeader(): bool
```

Comment:
I think this function is useless here.
The only usage you made of it is in this code :

```php
public static function isCountryFromBrowserAvailable(): bool
{
  $languageAvailable = static::getCountryIsoCodeFromHeader();
  if ($languageAvailable === null) {
    return false;
  }
  
  return Configuration::get('PS_DETECT_COUNTRY')
    && static::isCountryIsoCodeAvailableInHeader()
    && Validate::isLanguageIsoCode($languageAvailable);
}
``` 

Which is equivalent of : 
```php
public static function isCountryFromBrowserAvailable(): bool
{
  $languageAvailable = static::getCountryIsoCodeFromHeader();
  if ($languageAvailable === null) {
    return false;
  }
  
  return Configuration::get('PS_DETECT_COUNTRY')
    && (static::getCountryIsoCodeFromHeader() !== null)
    && Validate::isLanguageIsoCode($languageAvailable);
}
```

Has we can see, you already tested that `getCountryIsoCodeFromHeader();` return isn't null and you made two times this function call.

Reply:
Indeed some code is a bit redundant, but I wouldn't say the method is. Rather I would suggest doing this:

```php
    public static function isCountryFromBrowserAvailable(): bool
    {
        return Configuration::get('PS_DETECT_COUNTRY')
            && static::isCountryIsoCodeAvailableInHeader()
            && Validate::isLanguageIsoCode($languageAvailable);
    }
```

Reply:
Done differently ... but done